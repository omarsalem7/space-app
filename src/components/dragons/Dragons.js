import React from 'react';
import { useSelector } from 'react-redux';
import Dragon from '../dragon/Dragon';

const Dragons = () => {
  const dragons = useSelector((state) => state.dragonsReducer);
  return (
    <div>
      {dragons.map(({
        id, dragonName, description, image, reserved,
      }) => (
        <Dragon
          reserved={reserved}
          id={id}
          key={id}
          dragonName={dragonName}
          description={description}
          image={image}
        />
      ))}
    </div>
  );
};

export default Dragons;
