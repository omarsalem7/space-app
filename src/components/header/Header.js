import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../../assets/planet.png';
import './header.css';

const Header = () => {
  const linkStyle = {
    margin: '1rem',
    textDecoration: 'none',
    fontSize: '0.9rem',
    color: 'rgb(24, 159, 250)',
  };
  return (
    <div className="header-container">
      <ul className="header">
        <li>
          <img src={logo} alt="plant logo" className="img-logo" />
        </li>
        <li className="header-item"> Books Store</li>
      </ul>
      <ul className="header">
        <li>
          <NavLink style={linkStyle} to="/">
            Rockets
          </NavLink>
        </li>

        <li>
          <NavLink style={linkStyle} to="/missions">
            Missions
          </NavLink>
        </li>
        <li>|</li>
        <li>
          <NavLink style={linkStyle} to="/profile">
            My profile
          </NavLink>
        </li>
      </ul>
    </div>
  );
};

export default Header;
