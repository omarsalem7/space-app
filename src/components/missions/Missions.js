import React from 'react';
import { useSelector } from 'react-redux';
import { Table } from 'react-bootstrap';
import Mission from '../mission/Mission';

const Missions = () => {
  const missions = useSelector((state) => state.missionsReducer);
  return (
    <div className="missions">
      <Table striped bordered hover>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>

        {missions.map(({
          id, missionName, description, status,
        }) => (
          <Mission
            status={status}
            id={id}
            key={id}
            missionName={missionName}
            description={description}
          />
        ))}
      </Table>
    </div>
  );
};

export default Missions;
