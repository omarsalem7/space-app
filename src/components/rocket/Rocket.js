import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import './rocket.css';
import { Button, Badge } from 'react-bootstrap/';
import { reserveRocket, cancelReservation } from '../../redux/rockets/rockets';

const Rocket = ({
  id, rocketName, description, image, reserved,
}) => {
  const dispatch = useDispatch();
  return (
    <div className="rocket">
      <div className="rocket-img-container">
        <img className="rocket-img" src={image} alt={rocketName} />
      </div>
      <div className="rocket-detail">
        <h5>{rocketName}</h5>

        <p>
          {reserved && <Badge bg="info">Reserved</Badge>}
          {description}
        </p>

        {reserved ? (
          <Button
            onClick={() => dispatch(cancelReservation(id))}
            variant="outline-secondary"
            size="sm"
          >
            Cancel Reservation
          </Button>
        ) : (
          <Button
            onClick={() => dispatch(reserveRocket(id))}
            variant="primary"
            size="sm"
          >
            Reserve Rocket
          </Button>
        )}
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocketName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Rocket;
