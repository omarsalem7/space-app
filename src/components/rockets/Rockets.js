import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from '../rocket/Rocket';

const Rockets = () => {
  const rockets = useSelector((state) => state.rocketsReducer);
  return (
    <div>
      {rockets.map(({
        id, rocketName, description, image,
      }) => (
        <Rocket
          key={id}
          rocketName={rocketName}
          description={description}
          image={image[0]}
        />
      ))}
    </div>
  );
};

export default Rockets;
