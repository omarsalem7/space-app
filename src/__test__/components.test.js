import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Missions from '../components/missions/Missions';
import Rocket from '../components/rocket/Rocket';
import Rockets from '../components/rockets/Rockets';
import Header from '../components/header/Header';
import Dragons from '../components/dragons/Dragons';

describe('Test header component', () => {
  it('header renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('Test Missions renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Missions />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test Rocket renders correctly', () => {
  const item = {
    description: 'description',
    image: 'https://test.com',
    name: 'Rocket',
    id: 1,
    reserved: false,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Rocket
        description={item.description}
        image={item.image}
        rocketName={item.name}
        id={item.id}
        reserved={item.reserved}
      />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test Rockets renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test Dragons renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Dragons />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
