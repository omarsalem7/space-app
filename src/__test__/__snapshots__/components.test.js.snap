// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dragon renders correctly 1`] = `
<div
  className="rocket"
>
  <div
    className="rocket-img-container"
  >
    <img
      alt="Rocket"
      className="rocket-img"
      src="https://test.com"
    />
  </div>
  <div
    className="rocket-detail"
  >
    <h5>
      Rocket
    </h5>
    <p>
      description
    </p>
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Test Dragons renders correctly 1`] = `<div />`;

exports[`Test Missions renders correctly 1`] = `
<section
  className="missionSectionContainer"
>
  <li
    className="missionGridContainer"
  >
    <div
      className="headerText headerTextTopGridBlackBorder"
    >
      MISSIONS
    </div>
    <div
      className="headerText headerTextTopGridBlackBorder"
    >
      DESCRIPTION
    </div>
    <div
      className="headerText headerTextTopGridBlackBorder"
    >
      STATUS
    </div>
    <div
      className="headerText headerTextTopGridBlackBorder"
    />
  </li>
</section>
`;

exports[`Test Rocket renders correctly 1`] = `
<div
  className="rocket"
>
  <div
    className="rocket-img-container"
  >
    <img
      alt="Rocket"
      className="rocket-img"
      src="https://test.com"
    />
  </div>
  <div
    className="rocket-detail"
  >
    <h5>
      Rocket
    </h5>
    <p>
      description
    </p>
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Test Rocket renders correctly 2`] = `
<div
  className="rocket"
>
  <div
    className="rocket-img-container"
  >
    <img
      alt="Rocket"
      className="rocket-img"
      src="https://test.com"
    />
  </div>
  <div
    className="rocket-detail"
  >
    <h5>
      Rocket
    </h5>
    <p>
      description
    </p>
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Test Rockets List renders correctly 1`] = `<div />`;

exports[`Test Rockets renders correctly 1`] = `<div />`;

exports[`Test Rockets renders correctly 2`] = `<div />`;

exports[`Test header component header renders correctly 1`] = `
<div
  className="header-container"
>
  <ul
    className="header"
  >
    <li>
      <img
        alt="plant logo"
        className="img-logo"
        src="planet.png"
      />
    </li>
    <li
      className="header-item"
    >
       Books Store
    </li>
  </ul>
  <ul
    className="header"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(24, 159, 250)",
            "fontSize": "0.9rem",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(24, 159, 250)",
            "fontSize": "0.9rem",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className=""
        href="/dragons"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(24, 159, 250)",
            "fontSize": "0.9rem",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Dragons
      </a>
    </li>
    <li>
      |
    </li>
    <li>
      <a
        className=""
        href="/profile"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(24, 159, 250)",
            "fontSize": "0.9rem",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        My profile
      </a>
    </li>
  </ul>
</div>
`;

exports[`Test if Missions renders correctly 1`] = `
<section
  className="missionSectionContainer"
>
  <li
    className="missionGridContainer"
  >
    <div
      className="headerText headerTextTopGridBlackBorder"
    >
      MISSIONS
    </div>
    <div
      className="headerText headerTextTopGridBlackBorder"
    >
      DESCRIPTION
    </div>
    <div
      className="headerText headerTextTopGridBlackBorder"
    >
      STATUS
    </div>
    <div
      className="headerText headerTextTopGridBlackBorder"
    />
  </li>
</section>
`;

exports[`Test if Rocket renders correctly 1`] = `
<div
  className="rocket"
>
  <div
    className="rocket-img-container"
  >
    <img
      alt="Rocket"
      className="rocket-img"
      src="https://test.com"
    />
  </div>
  <div
    className="rocket-detail"
  >
    <h5>
      Rocket
    </h5>
    <p>
      description
    </p>
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Test if Rockets List renders correctly 1`] = `<div />`;
